### PR TITLE
[2.1 API] Enable printing deprecated warning info.

### PR DESCRIPTION
**Pull request by xiemoyuan**
### PR types
Bug fixes

### PR changes
APIs

### Describe
Enable printing deprecated warning info. Included:
```
python/paddle/dataset/cifar.py
python/paddle/dataset/conll05.py
python/paddle/dataset/flowers.py
python/paddle/dataset/imdb.py
python/paddle/dataset/imikolov.py
python/paddle/dataset/mnist.py
python/paddle/dataset/movielens.py
python/paddle/dataset/uci_housing.py
python/paddle/dataset/voc2012.py
python/paddle/dataset/wmt14.py
python/paddle/dataset/wmt16.py
python/paddle/fluid/dygraph/varbase_patch_methods.py
python/paddle/nn/__init__.py
python/paddle/nn/functional/loss.py
python/paddle/utils/deprecated.py
```

### CODE REVIEW

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: python/paddle/nn/__init__.py [hunk 1]
```diff
@@ -146,6 +146,7 @@
 @deprecated(
     since="2.0.0",
     update_to="paddle.nn.funcitional.diag_embed",
+    level=1,
     reason="diag_embed in paddle.nn will removed in future")
```
Comment (jzhang533) [inline python/paddle/nn/__init__.py]: will be removed

File: python/paddle/nn/__init__.py [hunk 2]
```diff
@@ -157,6 +158,7 @@ def diag_embed(*args):
 @deprecated(
     since="2.0.0",
     update_to="paddle.nn.utils.remove_weight_norm",
+    level=1,
     reason="remove_weight_norm in paddle.nn will removed in future")
```
Comment (jzhang533) [inline python/paddle/nn/__init__.py]: will be removed

File: python/paddle/nn/__init__.py [hunk 3]
```diff
@@ -168,6 +170,7 @@ def remove_weight_norm(*args):
 @deprecated(
     since="2.0.0",
     update_to="paddle.nn.utils.weight_norm",
+    level=1,
     reason="weight_norm in paddle.nn will removed in future")
```
Comment (jzhang533) [inline python/paddle/nn/__init__.py]: will be removed

File: python/paddle/nn/functional/loss.py [hunk 4]
```diff
@@ -1096,7 +1096,8 @@ def ctc_loss(log_probs,
     return loss_out
 
 
-@deprecated(since="2.0.0", update_to="paddle.nn.functional.cross_entropy")
+@deprecated(
+    since="2.0.0", update_to="paddle.nn.functional.cross_entropy", level=1)
```
Comment (jzhang533) [inline python/paddle/nn/functional/loss.py]: 是不是在reason参数里说明一下，`paddle.nn.functional.cross_entropy` 和 `softmax_with_cross_entropy`的行为是不一致的。

File: python/paddle/nn/__init__.py (hunk 1 above)
Comment (xiemoyuan) [reply python/paddle/nn/__init__.py]: Done. Thanks.

File: python/paddle/nn/__init__.py (hunk 2 above)
Comment (xiemoyuan) [reply python/paddle/nn/__init__.py]: Done. Thanks.

File: python/paddle/nn/__init__.py (hunk 3 above)
Comment (xiemoyuan) [reply python/paddle/nn/__init__.py]: Done. Thanks.

File: python/paddle/nn/functional/loss.py (hunk 4 above)
Comment (xiemoyuan) [reply python/paddle/nn/functional/loss.py]: Done. Thanks.

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
lgtm
